plugins/woocommerce/includes/class-wc-shipping-zone.php

```diff
@@ -459,4 +459,122 @@ public function delete_shipping_method( $instance_id ) {
 
 		return true;
 	}
+
+	/**
+	 * Get a specific shipping method by instance ID.
+	 *
+	 * @param int $instance_id Method instance ID.
+	 * @return WC_Shipping_Method|false Method instance or false if not found.
+	 */
+	public function get_shipping_method( $instance_id ) {
+		return WC_Shipping_Zones::get_shipping_method( $instance_id );
+	}
+
+	/**
+	 * Update a shipping method comprehensively.
+	 *
+	 * @param int   $instance_id Method instance ID.
+	 * @param array $data        Data to update (settings, enabled, order).
+	 * @return WC_Shipping_Method|\WP_Error Updated method instance or error.
+	 */
+	public function update_shipping_method( $instance_id, $data ) {
+		$method = $this->get_shipping_method( $instance_id );
+		if ( ! $method ) {
+			return new \WP_Error( 'woocommerce_rest_shipping_zone_method_invalid', __( 'Shipping method not found.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		global $wpdb;
+
+		// Update settings if provided.
+		if ( isset( $data['settings'] ) && is_array( $data['settings'] ) ) {
+			$result = $method->update_settings( $data['settings'] );
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		// Update method order if provided.
+		if ( isset( $data['order'] ) ) {
+			$this->set_method_order( $instance_id, absint( $data['order'] ) );
+		}
+
+		// Update enabled status if provided.
+		if ( isset( $data['enabled'] ) ) {
```

Comment:
When I don't send the `enabled` key when the `enabled` value is `false`, the updated data is returning `enabled` as `true`. Is this expected?

Reply:
Great catch! I dug into this and seems like this is a pre-existing bug. The issue is that `WC_Shipping_Zones::get_shipping_method()` was creating new instances of shipping method classes without loading the enabled and method_order properties from the database. This caused methods to always show enabled: 'yes' regardless of the actual database value. I have fixed this now and added a test too.